### DISCUSSION TITLE
Composite Keys

### DISCUSSION THREAD

**Discussion by asterissco** [orptech-com/laravel-migration-partition#36] [Q&A] [answered]
Hello,

Before all, thanks for this work. 

Well, when the table is created, it is mandatory to put primary keys. Why? I want to create a partition table with a simple primary key id and a date like rangePartitionKey but it is impossible with this solution

```        
Schema::createRangePartitioned('vessel_position_partition_version', function (Blueprint $table) {
            $table->id();
            $table->datetime('timestamp');
            $table->text('data')->nullable();
        }, null, null, 'timestamp');        
        
```

Thanks!

**Comment by ArdaKaraderi** [comment, marked as answer]
Hi, 

What you are trying to achieve is similar to PR #25. I didn't merge it because it only focused on the Range partition. If you can extend that solution to List and Hash partitions or arrange the extension with the PR author (@briets), I would be happy to personally review and merge all into the main branch for you.

Kind regards,
Arda
